plugins/chain_plugin/include/eosio/chain_plugin/chain_plugin.hpp

```diff
@@ -491,7 +491,7 @@ class read_only {
          if( p.lower_bound.size() ) {
             if( p.key_type == "name" ) {
                name s(p.lower_bound);
-               SecKeyType lv = convert_to_type<SecKeyType>( s.to_string(), "lower_bound name" ); // avoids compiler error
+               SecKeyType lv = convert_to_type<SecKeyType>( std::to_string(s.to_uint64_t()), "lower_bound name" ); // avoids compiler error
```

Comment:
look what you are doing here:
1. convert string to uint64 name. 
2. convert resulted uint64 to string.
3. convert resulted string to uint64.

I propose template straightforward solution to avoid that:
1. change declaration of `convert_to_type`:
```
template<typename Type, bool IsName=false>
Type convert_to_type(const string& str, const string& desc)

```

2. declare corresponding template specialization:
```
template<>
uint64_t convert_to_type<uint64_t, true>(const string& str, const string& desc);
```

implementation is something like this:
```
template<>
uint64_t convert_to_type<uint64_t, true>(const string& str, const string& desc) {
   try {
      return chain::string_to_uint64_t(str);
   } FC_RETHROW_EXCEPTIONS(warn, "Could not convert ${desc} name string '${str}' to key type.", ("desc", desc)("str",str) );
}
```

change line 494 (and same for 505) to:
`SecKeyType lv = convert_to_type<SecKeyType, true>( p.lower_bound, "lower_bound" );`

Reply:
I see where are you are going here. But, you can't specialize template functions the way you are wanting.  The "specialization" is from the return type.

I think a sufficient approach is to just create an overload that takes an `eosio::name` as the first parameter.

So,

``` c++
uint64_t convert_to_type(const eosio::name& n, const std::string& desc) {
   // to_uint64_t is basically noexcept
   return n.to_uint64_t();
}
```

Then in your caller, 
``` c++
SecKeyType lv = convert_to_type(s /*or name{p.lower_bound} */, "lower_bound");
```

Reply:
@larryk85 I tested it and it works. So at least clang permits this.
Your suggested alternative looks simpler than mine. Good idea.

Reply:
> @larryk85 I tested it and it works. So at least clang permits this.
> Your suggested alternative looks simpler than mine. Good idea.
> However it may need
> `constexpr if (is_same_v<uint64_t, SecKeyType>)` for compiler to compile.

1) Ah, I think I missed that you weren't trying to do partial specialization (didn't read it very well, my apologies).  I thought the pattern you were going with would've required partial specialization, I didn't pay attention to the defaulted value.

2) Yes, good call.  It will need an `if constexpr` wrapped around that call.  And an exception/error on the else branch, because it shouldn't be in that state.  If it did then something has gone awry.